gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFSInputStream.java

```diff
@@ -193,4 +203,16 @@ public int available() throws IOException {
   public IOStatistics getIOStatistics() {
     return streamStatistics.getIOStatistics();
   }
+
+  /**
+   * Verify that the input stream is open. Non blocking; this gives the last state of the volatile
+   * {@link #closed} field.
+   *
+   * @throws IOException if the connection is closed.
+   */
+  private void checkNotClosed() throws IOException {
+    if (closed) {
+      throw new IOException(gcsPath + ": " + FSExceptionMessages.STREAM_IS_CLOSED);
```

Comment:
should this be ClosedChannelException?

Reply:
Yes - I tried this - it doesn't take a string as an argument. That's why I decided to go ahead with IOException.